src/io/file_appender.rs

```diff
@@ -0,0 +1,10 @@
+use anyhow::Result;
+use arrow::record_batch::RecordBatch;
+use async_trait::async_trait;
+
+#[async_trait]
+pub trait FileAppender {
```

Comment:
I prefer to remove this trait first until we have more than one format to support. It's much easier to work with `ParquetWriter` directly at PoC stage.